### PR TITLE
Large update focused on fixing linting, tests browser refactoring, dependency updates

### PR DESCRIPTION
**Pull request by Swiftwork**
This is a pull-request focusing on improving the maintainability of the project. Preferably this should be merged into development when ready, but the branch is behind master at the moment.

**Refactoring**
- [x] Remove browser related code
- [x] Remove browser documentation

**Linting**
- [x] Remove deprecated tslint and migrate to eslint
- [x] Correct all linting errors

**Documentation**
- [x] Add new examples. **edit:** more examples needed, but docs space created
- [x] Refer browser usage to [new browser project](https://github.com/zxing-js/browser)

**Testing**
- [x] ~~Migrate to a new test tool without browser requirements~~ mocha remained with ts-mocha wrapper and karma removed
- [ ] ~~Correct all failing tests~~ out of scope for this PR, but added logging levels and improved clarity

**Dependencies**
- [x] Update dependencies to latest version

**Building**
- [x] Streamline build process
- [ ] Configure CI/CD
- [ ] Enable cross-platform builds

### CODE REVIEW

**Comment by Swiftwork** [comment]
Using TypeDoc with plugins to generate and improve documentation for the library. An example can be seen on this fork's [github pages](https://swiftwork.github.io/zxing-js-library).

**Comment by Swiftwork** [comment]
@odahcam when I have time send me a message on the [ZXing-js Gitter](https://gitter.im/zxing-js/library) and we can discuss what in this pull-request should be reverted and how we should proceed. I would like to see some of this work merged to the development branch soon. I'd like to help close some other issue and get this a little more production-ready for React Native 😃.

File: package.json [hunk 1]
```diff
@@ -55,9 +55,7 @@
     "type-check:watch": "yarn type-check -- --watch",
     "build": "yarn clean && rollup -c",
     "build:umd:min": "gzip index.min.js -c > index.min.js.gz",
-    "test": "yarn test:build && yarn test:run",
-    "test:build": "tsc --build tsconfig.test.json",
-    "test:run": "mocha -r tsconfig-paths/register --timeout 200000 output/tests/**/*.spec.js",
+    "test": "ts-mocha -p tsconfig.test.json --paths",
```
Comment (odahcam) [inline package.json:58]: We can proceed with ts-mocha as long as it allows us to debug the tests.

**Comment by odahcam** [comment]
I liked the changes, do you still wanna commit something before changing the "Draft" status of this PR?

File: package.json (hunk 1 above)
Comment (Swiftwork) [reply package.json:58]: That is something we should evaluate, need to know how you debug them so that we can test it :)

**Comment by Swiftwork** [comment]
Mainly draft so that we have a chance to discuss it first. There are some changes there that would be good to talk through. Also I'd like to see a merge to development before master :)

File: package.json (hunk 1 above)
Comment (odahcam) [reply package.json:58]: Mainly via VSCode debugger. I do now that ts-mocha allows us to debug but currently at the cost of pointer precision (it seems the source maps are wrongly generated everytime) and some performance, but using it wouldn't be a problem if at least the source maps work fine. Also, will this impact the current CI? Current CI is already very slow and I'd not like to get it slower 🐌

Comment (odahcam) [reply package.json:58]: In short: debugging tests is already kinda broken, so VSCode debugging won't be a problem.

**Pull request merged**